otter/convergence/service.py

```diff
@@ -172,7 +172,8 @@ def update_cache(group, servers, lb_nodes, now):
         sd = thaw(server.json)
         if is_autoscale_active(server, lb_nodes):
             sd["_is_as_active"] = True
-        server_dicts.append(sd)
+        if server.state != ServerState.DELETED:
```

Comment:
Not quite. The logic in each method (before) did two things: One doesn't add the `_is_as_active` _key_ if that check doesn't pass. The other _doesn't add the server to the cache_ if it's deleted. That's not the same thing and that's why both are represented here. We want (I'm assuming) that deleted servers shouldn't be in the cache at all, not just have that key omitted.